### PR TITLE
EOS-23527: SSPL post_install failed on network validation during CORTX deployment

### PR DESCRIPTION
**Pull request by mariyappanp**
…X deployment

## Coding:

* [x] Did you follow coding standard and verified it with flake8? 

* [x] Did your address all Codacy issues?

## Testing:

* [x] Unittest updated and all unittests are passing?

* [x] Sanity/Self tests are updated (if applicable)?

* [x] Manual testing done covering happy path and non-happy path?

## Integration:

* [x] If there is any interface change, did you communicate it to other components?
* [x] If there is any interface change, other componenent gate gatekeepers are ready to accept the change?
* [x] Did you complete deployment test if applicable?

## PR checklist:
* [x] Did you add Jira number to the PR?

* [x] DCO and cla-signed?

### CODE REVIEW

**Review by shriya-deshmukh — APPROVED** [review]

File: low-level/files/opt/seagate/sspl/setup/sspl_post_install.py [hunk 1]
```diff
@@ -103,7 +104,17 @@ def validate(self):
             # cluster_id required for decrypting the secret is only available from
             # the prepare stage. However accessibility validation will be done in
             # prepare stage. So at this time, validating ip reachability is fine.
-            NetworkV().validate("connectivity", [bmc_ip, primary_ip, secondary_ip])
+            for ip in [bmc_ip, primary_ip, secondary_ip]:
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_post_install.py:107]: I thought we decided to extend validators itself, taking these retry count, sleep time as inputs. So others can also take advantage !?
For quick fix in SSPL, this is fine obviously

Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_post_install.py:107]: `ping -c 1 -W 1 <ip>` is the cmd used in ip connectivity check.
Even if we increase ping `-c 1` to `-c 50`, exception will be thrown for one of the packet loss.
So I feel retry from SSPL side is better here.

Comment (sumedhak27) [reply low-level/files/opt/seagate/sspl/setup/sspl_post_install.py]: Hi @mariyappanp, there is one option `-w (small w not capital)` which is a deadline, so for example we can have a deadline of 4sec `-w 4` and count of 2 `-c 2`. for this config ping will wait for 4 sec to get 2 packets successfully sent, if 2 pkts are sent before the deadline then ping will succeed even if 1,2 or 3, etc pkts are dropped.

Is it possible to use this❔

Comment (sumedhak27) [reply low-level/files/opt/seagate/sspl/setup/sspl_post_install.py]: image

Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_post_install.py]: Possible in cortx-utils. But still SSPL will have minor change to pass it.
So two PR will be needed in that case.

Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_post_install.py]: @sumedhak27 , current approach with SimpleProcess in cortx-utils is same.
If return code is not 0, its reachable. If non-zero, then unreachable.

Comment (cdeshmukh) [reply low-level/files/opt/seagate/sspl/setup/sspl_post_install.py]: yes agree @mariyappanp , we will go ahead with changes in SSPL itself right now

**Review by cdeshmukh — APPROVED** [review]

**Review by sumedhak27 — APPROVED** [review]

**Pull request merged**